### PR TITLE
non-blocking td_run> operator

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDJobOperator.java [hunk 1]
```diff
@@ -202,4 +202,25 @@ public ArrayValue next()
...
+                String message = job.getCmdOut() + "\n" + job.getStdErr();
+                throw new TaskExecutionException(message, buildExceptionErrorConfig(ex));
+            }
+            catch (Exception getJobInfoFailed) {
+                getJobInfoFailed.addSuppressed(ex);
+                throw Throwables.propagate(getJobInfoFailed);
+            }
+        }
+        catch (InterruptedException ex) {
+            throw Throwables.propagate(ex);
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDJobOperator.java]: Restore the interrupted status

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TdLoadOperatorFactory.java [hunk 2]
```diff
@@ -176,27 +176,6 @@ private TaskResult processJobResult(TDOperator op, TDJobOperator j, TDJobSummary
                     .build();
         }
 
-        private TDJobSummary checkJobStatus(TDJobOperator j)
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TdLoadOperatorFactory.java]: 👍

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TdRunOperatorFactory.java [hunk 3]
```diff
@@ -42,44 +53,84 @@ public Operator newTaskExecutor(Path workspacePath, TaskRequest request)
...
 
-            try (TDOperator op = TDOperator.fromConfig(params)) {
-                TDJobOperator j = op.startSavedQuery(name, Date.from(sessionTime));
-                logger.info("Started a saved query name={} with time={}", name, sessionTime);
+                // Start the job
+                if (!existingJobId.isPresent()) {
+                    TDJobOperator j = op.startSavedQuery(name, Date.from(sessionTime), existingDomainKey.get());
+                    logger.info("Started a saved query name={} with time={}", name, sessionTime);
+                    state.set(JOB_ID, j.getJobId());
+                    state.set(POLL_ITERATION, 1);
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TdRunOperatorFactory.java]: [minor] Maybe we can remove this line

**Comment by komamitsu** [comment]
Added some comments. Other than that look good!

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TdRunOperatorFactory.java (hunk 3 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TdRunOperatorFactory.java]: Yeah, I want to do that in a separate PR that factors out all the duplicated non-blocking job execution logic from the td operators into reusable code.

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDJobOperator.java (hunk 1 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDJobOperator.java]: Yeah, unfortunately there's a lot of similar code in digdag where we're not properly restoring interrupted status. We should fix that everywhere.
